### PR TITLE
user_booking_ticket

### PR DESCRIPTION
**Pull request by MartaSolo**
- booking ticket functionality for a user with styling for desktop, mobile;
- plus small changes in other components / utils etc.
step 1 - desktop
step 2 - dekstop
step 2 - mobile
step 3 - desktop
step 3 - mobile

### CODE REVIEW

File: src/components/booking/BookingSuccess.vue [hunk 1]
```diff
@@ -0,0 +1,138 @@
+<script setup lang="ts">
+import { computed, watch } from "vue";
+import { storeToRefs } from "pinia";
+import { formattedDateAndTime } from "@/utils/formattedDate.js";
+import { useReservationStore } from "@/stores/reservation.js";
+import LoadingData from "@/components/global/LoadingData.vue";
+import ErrorMessage from "@/components/global/LoadingData.vue";
+import BookingSuccessTitle from "@/components/booking/BookingSuccessTitle.vue";
+
+import BaseButton from "../global/BaseButton.vue";
```
Comment (szymonlicau) [inline src/components/booking/BookingSuccess.vue]: We should stick to using the aliases everywhere 😄

```suggestion
import BaseButton from "@/components/global/BaseButton.vue";
```

File: src/components/booking/BookingSuccess.vue [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  reservedSeance,
+  reservation,
+  reservationIsLoading,
+  reservationError,
+  getReservationErrorMessage,
+} = storeToRefs(reservationStore);
+
+const getReservationId = computed(() => {
+  return reservedSeance.value?.id;
+});
```
Comment (szymonlicau) [inline src/components/booking/BookingSuccess.vue]: Why not simply name it `reservationId`? Computed is not a function in itself so the `get` part is kind of strange 😄 
```suggestion
const reservationId = computed(() => {
  return reservedSeance.value?.id;
});
```

File: src/components/booking/BookingSuccess.vue [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+const getReservationData = () => {
```
Comment (szymonlicau) [inline src/components/booking/BookingSuccess.vue]: Same with this one
```suggestion
const reservationData = () => {
```

File: src/components/booking/BookingSuccess.vue [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+watch(getReservationId, (newValue, oldValue) => {
+  getReservationData();
+});
+
+const getTicketsNumber = computed(() => {
+  return reservation.value?.tickets.length;
+});
+
+const getSubtitle = () => {
```
Comment (szymonlicau) [inline src/components/booking/BookingSuccess.vue]: And these two

File: src/components/booking/ChooseSeats.vue [hunk 5]
```diff
@@ -0,0 +1,126 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import { storeToRefs } from "pinia";
+import { useReservationStore } from "@/stores/reservation.js";
+import BaseButton from "@/components/global/BaseButton.vue";
+
+interface chosenSeat {
```
Comment (szymonlicau) [inline src/components/booking/ChooseSeats.vue]: We should write our types in `PascalCase` to let them stand out and be consistent 😄 
```suggestion
interface ChosenSeat {
```

File: src/components/booking/ChooseTickets.vue [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+const router = useRouter();
+
+const reservedSeatsTickets: Ref<ReservedSeatTicket[]> = ref(
+  reservedSeats.value.map((reservedSeat) => {
+    return { seat: reservedSeat, ticket_type_id: 1 };
+  })
+);
+
+const terms: Ref<boolean> = ref(false);
```
Comment (szymonlicau) [inline src/components/booking/ChooseTickets.vue]: No need to be *that* explicit - if we pass a boolean value, TypeScript will know everything :)
```suggestion
const terms = ref(false);
```

File: src/components/booking/ChooseTickets.vue [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+};
+
+const bookTickets = () => {
+  if (isUserLoggedIn.value) {
+    reservationStore.reserveSeance(reservedSeatsTickets.value);
+    changeStep(3);
+  }
+  if (!isUserLoggedIn.value) {
+    router.push({ name: "UserLogIn" });
+  }
```
Comment (szymonlicau) [inline src/components/booking/ChooseTickets.vue]:
```suggestion
  } else {
    router.push({ name: "UserLogIn" });
  }
```

File: src/i18n.ts [hunk 8]
```diff
@@ -4,20 +4,20 @@ import en from "@/locales/en";
 import pl from "@/locales/pl";
 
 export default createI18n({
-  locale: "en",
-  fallbackLocale: "en",
+  locale: "en-GB",
+  fallbackLocale: "en-GB",
```
Comment (szymonlicau) [inline src/i18n.ts:8]: Why the change from `en` to `en-GB`? 🤔

File: src/pages/BookTicket.vue [hunk 9]
```diff
@@ -1,19 +1,22 @@
...
+import ChooseSeats from "@/components/booking/ChooseSeats.vue";
+import ChooseTickets from "@/components/booking/ChooseTickets.vue";
+import BookingSuccess from "@/components/booking/BookingSuccess.vue";
 
 const props = defineProps<{
   movieSeanceId: String;
 }>();
 
-const step = ref(1);
+const step: Ref<number> = ref(1);
```
Comment (szymonlicau) [inline src/pages/BookTicket.vue]: TypeScript should infer that one properly 🤔 
```suggestion
const step = ref(1);
```

File: src/utils/hall.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+export const numberOfSeatsInRow = (array) => {
+  const listOfSeats = array.map((el) => el.seatNumber);
+  const uniqueSeats = listOfSeats.filter(
+    (seat, i, arr) => arr.indexOf(seat) === i
+  );
+  return uniqueSeats.length;
+};
+
+export const hallPlan = (array) => {
```
Comment (szymonlicau) [inline src/utils/hall.js]: Likewise for these function, argument names like `array` are too generic! We should give more hints to what's going on here 😄

**Review by szymonlicau — CHANGES_REQUESTED** [review]

File: src/i18n.ts (hunk 8 above)
Comment (MartaSolo) [reply src/i18n.ts:8]: to get data format DD/MM/YYYY

**Review by szymonlicau — APPROVED** [review]

**Pull request merged**